src/main/scala/ai/privado/languageEngine/ruby/cache/PackageTable.scala

```diff
@@ -0,0 +1,45 @@
+package ai.privado.languageEngine.ruby.cache
+
+import io.joern.x2cpg.Defines.DynamicCallUnknownFullName
+
+import java.util.concurrent.ConcurrentHashMap
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+case class MethodTableModel(methodName: String, parentClassPath: String, classType: String, methodFullName: String)
+
+case class PackageContext(moduleName: String, packageTable: PackageTable)
+
+class PackageTable() {
+
+  private val methodTableMap = new ConcurrentHashMap[String, mutable.HashSet[MethodTableModel]]()
+
+  def addPackageMethod(moduleName: String, methodName: String, parentClassPath: String, classType: String): Unit = {
+    val packageMethod =
+      MethodTableModel(methodName, parentClassPath, classType, s"$moduleName::program:${parentClassPath}:$methodName")
```

Comment:
change methodFullName to `s"$moduleName::program.${parentClassPath}.$methodName"` We will be using `.` as a seperator